tools/Azure.Mcp.Tools.Foundry/src/Services/FoundryService.cs

```diff
@@ -1688,4 +1673,54 @@ public string SerializeToolDefinition()
             return Encoding.UTF8.GetString(bytes.GetBuffer(), 0, (int)bytes.Length);
         }
     }
+
+    internal static Microsoft.Extensions.AI.ChatMessage CreateRequestMessage(string toolCallId, string name, Dictionary<string, object?> arguments, RunStep step)
+    {
+        return new(ChatRole.Assistant, [new FunctionCallContent(toolCallId, name, arguments)])
+        {
+            AuthorName = step.AssistantId,
+            MessageId = step.Id,
+            RawRepresentation = step,
+        };
+    }
+
+    internal static Microsoft.Extensions.AI.ChatMessage CreateResponseMessage(string toolCallId, List<AgentFileSearchResult> result, RunStep step)
+    {
+        return new(ChatRole.Tool, [new FunctionResultContent(toolCallId, result)])
+        {
+            AuthorName = step.AssistantId,
+            MessageId = step.Id,
+            RawRepresentation = step,
+        };
+    }
+
+    internal static Microsoft.Extensions.AI.ChatMessage CreateResponseMessage(string toolCallId, JsonElement result, RunStep step)
```

Comment:
Do we need all these different static methods given `FunctionResultContent`'s constructor is `string callId, object? result)`. Cant we just have:

```suggestion
    internal static Microsoft.Extensions.AI.ChatMessage CreateResponseMessage(string toolCallId, object? result, RunStep step)
```